src/redux/activities/thunks.js

```diff
@@ -1,13 +1,92 @@
 import * as actions from './actions';
 
 export const getActivities = async (dispatch) => {
+  dispatch(actions.getActivitiesPending());
   try {
-    dispatch(actions.getActivitiesPending());
     const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/`);
     const data = await res.json();
-    dispatch(actions.getActivitiesPending());
-    dispatch(actions.getActivitiesSuccess(data.data));
+    dispatch(actions.resetInitialState());
+
+    if (res.status === 200) {
+      dispatch(actions.getActivitiesSuccess(data.data));
+    }
+
+    if (res.status !== 200) {
+      throw new Error(data.message);
+    }
+  } catch (error) {
+    dispatch(actions.getActivitiesError(error.message));
+  }
+};
+
+export const postActivity = async (dispatch, newActivity) => {
+  dispatch(actions.postActivitiesPending());
+  try {
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/activities/`, {
+      method: 'POST',
+      body: JSON.stringify(newActivity),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+    const data = await res.json();
+    dispatch(actions.resetInitialState());
+
+    if (res.status === 201) {
+      dispatch(actions.postActivitiesSuccess(data.data, data.message));
+    }
+
+    if (res.status !== 201) {
+      throw new Error(data.message);
```

Comment:
Im not sure this are doing what they are supposed to, I tried them on my end and whenever the res was not ok the error would just get caught in the catch and never get thrown in the if statement

Reply:
If the error was caught in the catch its because the throw error is working, thats the throw error function... If it wasn't working, the modal wouldn't say the error message